### PR TITLE
Add CompileDesignTime for returning compiler arguments

### PR DESCRIPTION
**Pull request by davkean**
For compatibility with the legacy project system, we call Compile target - however, this does not return the command-line arguments. Introduce a target that explicitly returns them, and in line with other XXXDesignTime targets.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Build/BuildProperty.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{
+    /// <summary>
+    ///     Contains common MSBuild build properties.
+    /// </summary>
+    internal static class BuildProperty
+    {
+        /// <summary>
+        ///     Indicates whether CoreCompile target should skip compiler execution completely.
+        /// </summary>
+        public static string SkipCompilerExecution = nameof(SkipCompilerExecution);
```
Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Build/BuildProperty.cs]: Make these readonly

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Build/BuildProperty.cs]: Whoops they are supposed to be constant.

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Build/BuildProperty.cs]: Marked them constant.

File: src/Targets/Microsoft.CSharp.DesignTime.targets [hunk 2]
```diff
@@ -43,4 +43,19 @@
...
+  
+  <!-- Targets -->
+
+  <!-- Returns Csc command-line arguments for the language service -->
+  <Target Name="CompileDesignTime"
+          Returns="@(CscCommandLineArgs)"
+          DependsOnTargets="Compile"
+          >
+
+    <Error Condition="'$(SkipCompilerExecution)|$(ProvideCommandLineArgs)' == 'true|true'"
```
Comment (srivatsn) [inline src/Targets/Microsoft.CSharp.DesignTime.targets]: Is the condition correct? Shouldn't it error in the opposite case when they aren't true|true?

Comment (davkean) [inline src/Targets/Microsoft.CSharp.DesignTime.targets]: Yes you are right. Interestingly enough, this isn't failing the build - need to figure out why...

Comment (davkean) [inline src/Targets/Microsoft.CSharp.DesignTime.targets]: Oh, I see it now. :(

Comment (davkean) [inline src/Targets/Microsoft.CSharp.DesignTime.targets]: Fixed. The project fault that I just fixed in CPS - caused me not to see this.

**Pull request merged**